### PR TITLE
EknMediaBin: implemented basic audio mode.

### PR DESCRIPTION
**Pull request by xjuan**
Added audio-mode property

https://phabricator.endlessm.com/T14204

### CODE REVIEW

**Comment by tchx84** [comment]
I tried audio player with a real app and it looks OK, but I keep seeing the following error:

```
Gdk-CRITICAL **: gdk_window_set_cursor: assertion 'GDK_IS_WINDOW (window)' failed
```

Also tried testing this with the example player doing:

```
diff --git a/examples/eos-player.c b/examples/eos-player.c
index 03d8423..c936bb7 100644
--- a/examples/eos-player.c
+++ b/examples/eos-player.c
@@ -39,6 +39,7 @@ main (int argc, char **argv)
 
   bin = ekn_media_bin_new ();
   ekn_media_bin_set_uri (EKN_MEDIA_BIN (bin), argv[1]);
+  ekn_media_bin_set_audio_mode (EKN_MEDIA_BIN (bin), TRUE);
   gtk_container_add (GTK_CONTAINER (window), bin);
 
   /* Update window title */
```

But it looks messed up:

screenshot from 2016-12-28 10-22-26

**Comment by xjuan** [comment]
Yes I ned to fix that warning, even tough is unrelated to this task
and the glitch only happens on a VM

File: lib/eosknowledgeprivate/ekn-media-bin.c [hunk 1]
```diff
@@ -686,17 +691,20 @@ ekn_media_bin_fullscreen_apply (EknMediaBin *self, gboolean fullscreen)
...
       gtk_container_remove (GTK_CONTAINER (priv->fullscreen_window), priv->overlay);
-      gtk_container_add (GTK_CONTAINER (self), priv->overlay);
+      gtk_container_add (GTK_CONTAINER (priv->stack), priv->overlay);
+      gtk_stack_set_visible_child (GTK_STACK (priv->stack), priv->overlay);
 
       gtk_widget_destroy (GTK_WIDGET (priv->fullscreen_window));
       g_clear_object (&priv->fullscreen_window);
 
       gtk_image_set_from_icon_name (priv->fullscreen_image, EMB_ICON_NAME_FULLSCREEN, EMB_ICON_SIZE);
+
```
Comment (tchx84) [inline lib/eosknowledgeprivate/ekn-media-bin.c]: nit: extra spaces here

**Comment by tchx84** [comment]
One more thing, If I set the player in audio mode but play a video, a separate window appears with the video... which looks kind of odd.

File: lib/eosknowledgeprivate/ekn-media-bin.c [hunk 2]
```diff
@@ -1666,14 +1751,17 @@ ekn_media_bin_init_playbin (EknMediaBin *self)
...
  *
  * Returns a new #EknMediaBin
  *
  */
 GtkWidget *
-ekn_media_bin_new ()
+ekn_media_bin_new (gboolean audio_mode)
 {
-  return (GtkWidget*) g_object_new (EKN_TYPE_MEDIA_BIN, NULL);
+  return (GtkWidget*) g_object_new (EKN_TYPE_MEDIA_BIN,
```
Comment (tchx84) [inline lib/eosknowledgeprivate/ekn-media-bin.c]: nit: one extra space at the end of this line.

**Comment by tchx84** [comment]
Works great now!

I think I will merge it together with the other PR (so I can test it in a real app), also there is one small nit in case you can fix it.

**Pull request merged**